src/BeneficiariesFile.hs

```diff
@@ -50,16 +51,19 @@ parseBeneficiary conf = toBeneficiary . words
       makeBeneficiary addr (maybeToMissing "quantity" conf.dropAmount) (maybeToMissing "assetclass" conf.assetClass)
     toBeneficiary _ = Left "Invalid number of inputs"
 
-    makeBeneficiary :: Text -> Either Text Integer -> Either Text AssetClass -> Either Text Beneficiary
-    makeBeneficiary addr eAmt eAc = Beneficiary <$> parseAddress conf.usePubKeys addr <*> eAmt <*> eAc
+    scaleAmount :: Ratio Integer -> Integer
+    scaleAmount r = numerator r * conf.decimalPlaces `div` denominator r
```

Comment:
does multiplying the numerator by the decimal places do what we want? should it not be something like `10^decimal places`?